opal/corelib/runtime.js

```diff
@@ -1321,7 +1321,7 @@
       // A splatted array must be copied
       return value.slice();
     }
-    else if (value['$respond_to?']('to_a', true)) {
+    else if (typeof(value.$to_a) === 'function' && !value.$to_a.$$stub) {
```

Comment:
This code is not 100% correct for MRI. It **must** call `respond_to?`. I think it could be replaced to
``` js
else if (
  (
    // checking through 'respond_to?' if possible
    typeof(value['$respond_to?']) === 'function' &&
    !value['$respond_to?'].$$stub &&
    value['$respond_to?']('to_a', true)
  ) || (
    // we are in 'opal/mini', checking directly through 'to_a'
    typeof(value.$to_a) === 'function' &&
    !value.$to_a.$$stub
  )
) {
  // calling to_a
}
```

There are 4 occurrences of `respond_to?` in `runtime.js`, so it could be extracted to something like
``` js
Opal.respond_to = function(method_name, include_private) {
  // the code here
}
```
And also we could make some flag on `Opal.config` (js object) that indicates whether we running on `opal/full` or `opal/mini`, it can simplify the logic.